metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala

```diff
@@ -90,7 +91,13 @@ final class ImplementationProvider(
         params,
         currentDocument
       )
-      symbolOccurrence <- positionOccurrence.occurrence.toIterable
+      symbolOccurrence <- {
```

Comment:
Later when we look for the implementation we load semanticDB for the file and find occurence in `textDocument.occurences`, can the same situation happen there? If the annotation is on the implementing class?

Reply:
ah, I think so. The occurrence of the sealed trait is not a "definition" anymore in that case (it's a "reference" IIRC). Can you think of a test case for stressing it?

Reply:
That could be a modified test case in this PR I think:
```
    """|/a/src/main/scala/a/Main.scala
       |package a
       |import io.circe.generic.JsonCodec
       |trait Be@@ing
       |@JsonCodec sealed trait <<Animal>> extends Being
       |object Animal {
       |  case object <<Dog>> extends Animal
       |  case object <<Cat>> extends Animal
       |}
       |""".stripMargin
```

this most likely will not find `Animal`

Reply:
you're right, I've added the (now failed) test, so that we can work on this tomorrow.

Reply:
ok, I've pushed a fix for that case too. There's indeed a lot of repetion and I hope we can come up with a more harmonized strategy for finding a definition of a symbol in the next days.

Reply:
Thanks @gabro !